### DISCUSSION TITLE
Can OWL be "trained" to locate and identify specific colors and/or textures?

### DISCUSSION THREAD

**Discussion by RayElkins719** [geezacoleman/OpenWeedLocator#121] [Q&A]
Looking for a solution to use drone-mounted or hand-held camera system as a second (and sharper) set of eyes for foraging. I've been researching how I can possibly build a system to do it and landed here. I read that algorithms exist that allow OWL to identify specific textures and colors. I'm not looking for something that makes a positive identification, just inquiring as to whether it is possible to build a system that provides a way to cover more ground faster, can reach harder to access locations, and alerts when a specific color and/or texture is spotted.
 I've built and written code with Arduino but the Raspberry Pi and recognition stuff is new for me.
Thanks!!

**Comment by geezacoleman** [comment]
Hi Ray,

Thanks for the question! In its most basic form the OWL just looks for a colour, which is currently set to green - to find the green weeds in fallow. Do you know how diverse/difficult your foraging will be or do you have any examples? Perhaps if the thing you're after is clearly different from the background you could just change the detection thresholds here. We have a couple of different algorithms to choose from.

You could also train object detection models or image classification models to predict the presence/absence of a forageable (?) things too. This then requires a bit more hardware/software/data but is also doable. Then there are more basic models like SVMs or RandomForests etc that could also work. Just depends on the situation.

Not sure how helpful this is, but happy to try answer further questions if needed!

**Reply by RayElkins719** [reply]
Thank you for the reply! Sorry so slow getting back to you. My first thought is morel and "chicken of the woods" mushrooms since the texture of the morel is very unique on the forest floor and the bright color of the chicken is easily spotted. I'm optimistic that if I could get it to alert on these (especially the texture of the morel), I could maybe tweak and play with it to recognize other mushrooms (or at least alert on the color or texture, to save me a lot of steps.
The morel is fairly easy to recognize in the woods but it takes a lot of walking to actually run across them. Obviously the chicken's color would be the easier to identify against the colors in the forest, or at least I'm assuming. Am I far off?


![434872106_2173329469678394_5146083160442709638_n](https://github.com/geezacoleman/OpenWeedLocator/assets/165503496/ec59bed3-7ae4-4ac5-9988-807e46499a08)
![435181328_729660829366937_483300143843929961_n](https://github.com/geezacoleman/OpenWeedLocator/assets/165503496/5096ccc7-3e6c-468b-9524-9e246204affc)
![435061394_852561850041270_8060764531435047039_n](https://github.com/geezacoleman/OpenWeedLocator/assets/165503496/b1519ea6-ac92-4ab8-b1d5-0f5e0e2c96ef)
